screens/ReservationCalendarScreen.js

```diff
@@ -0,0 +1,508 @@
+import {Calendar, LocaleConfig, Agenda, AgendaList} from 'react-native-calendars';
+import {api, dispatchFetchRequest} from '../constants/Backend'
+import {NavigationEvents} from 'react-navigation'
+import TimeZoneService from "../helpers/TimeZoneService";
+import moment from "moment-timezone";
+import {StyledText} from '../components/StyledText'
+import {RenderAgenda, DayCalendar, ReservationDayEvent, ReservationEvent} from "../components/Calendars";
+import AddBtn from '../components/AddBtn'
+import Modal from 'react-native-modal';
+import {OptionModal} from "../components/OptionModal";
+import {MainActionButton} from '../components/ActionButtons'
+import SegmentedControlTab from "react-native-segmented-control-tab";
+import {CheckBox} from 'react-native-elements'
+import Icon from 'react-native-vector-icons/Ionicons'
+import {CalendarEvent} from "../components/CalendarEvent";
+import {Ionicons} from '@expo/vector-icons';
+import {compose} from "redux";
+import {connect} from 'react-redux'
+import {MaterialCommunityIcons} from '@expo/vector-icons';
+import {getTableLayouts, getTableLayout} from '../actions'
+
+class CalendarScreen extends React.Component {
+    static navigationOptions = {
+        header: null
+    }
+    static contextType = LocaleContext
+
+    constructor(props, context) {
+        super(props, context)
+        const timezone = TimeZoneService.getTimeZone()
+        this.state = {
+            isLoading: true,
+            calendarMode: 'month',
+            selectedDate: moment(new Date()).tz(timezone).format("YYYY-MM-DD"),
+            users: [],
+            isShowModal: false,
+            calendarTypeIndex: 0,
+            searchTypeIndex: 1,
+            modalTasks: [],
+            nowYear: new Date().getFullYear(),
+            nowMonth: new Date().getMonth() + 1,
+            monthLoading: false,
+            reservationEvents: [],
+            reservationEventsByDate: [],
+            showReservationFormModal: false,
+        }
+        context.localize({
+            en: {
+                calendar: {
+                    monthly: 'Monthly',
+                    daily: 'Daily',
+                    reservation: 'Reservation',
+                    calendarOption: 'Calendar Option',
+                }
+            },
+            zh: {
+                calendar: {
+                    monthly: '月曆',
+                    daily: '日曆',
+                    reservation: '訂位',
+                    calendarOption: '日程表選項',
+                }
+            }
+        })
+        LocaleConfig.locales['zh-Hant-TW'] = {
+            monthNames: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+            monthNamesShort: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+            dayNames: ['週日', '週一', '週二', '週三', '週四', '週五', '週六'],
+            dayNamesShort: ['週日', '週一', '週二', '週三', '週四', '週五', '週六'],
+            today: '今日'
```

Comment:
What is this for?

Reply:
Calendar component translation, it's different with t